From cbdecd8e6dd87ff1aa8dce3c4c7e1a2c2f259cb0 Mon Sep 17 00:00:00 2001
From: robinz1x <robinx.zhang@intel.com>
Date: Thu, 19 Jul 2018 15:13:44 +0800
Subject: [PATCH] Check whether guest user is already created or not

Car settings part:
Check whether the guest user is created or not when
click "Switch" in GuestFragment or "Guest" icon in
UserGridRecylerView.

If guest user already created, switch to it;
otherwise, create a new guest user.

Change-Id: I98e2b219d3ebfc66042db04c9a527fefe27e8c8a
Tracked-On: OAM-72293
Signed-off-by: robinz1x <robinx.zhang@intel.com>
Signed-off-by: Yan, WalterX <walterx.yan@intel.com>
---
 src/com/android/car/settings/users/GuestFragment.java        | 9 ++++++++-
 src/com/android/car/settings/users/UserGridRecyclerView.java | 8 +++++++-
 2 files changed, 15 insertions(+), 2 deletions(-)

diff --git a/src/com/android/car/settings/users/GuestFragment.java b/src/com/android/car/settings/users/GuestFragment.java
index eaec0d5..aa9c95e 100644
--- a/src/com/android/car/settings/users/GuestFragment.java
+++ b/src/com/android/car/settings/users/GuestFragment.java
@@ -17,6 +17,7 @@
 package com.android.car.settings.users;
 
 import android.car.user.CarUserManagerHelper;
+import android.content.pm.UserInfo;
 import android.graphics.drawable.Drawable;
 import android.os.Bundle;
 import android.view.View;
@@ -75,7 +76,13 @@ public class GuestFragment extends ListItemSettingsFragment {
         switchUserBtn.setText(R.string.user_switch);
         switchUserBtn.setOnClickListener(v -> {
             getActivity().onBackPressed();
-            mCarUserManagerHelper.startNewGuestSession(getContext().getString(R.string.user_guest));
+            UserInfo currentGuestUser = mCarUserManagerHelper.findCurrentGuestUser();
+            // If guest user already created, switch to it; otherwise, create a new guest user
+            if (currentGuestUser != null) {
+                mCarUserManagerHelper.switchToUser(currentGuestUser);
+            } else {
+                mCarUserManagerHelper.startNewGuestSession(getContext().getString(R.string.user_guest));
+            }
         });
     }
 
diff --git a/src/com/android/car/settings/users/UserGridRecyclerView.java b/src/com/android/car/settings/users/UserGridRecyclerView.java
index b6d73fa..a31813d 100644
--- a/src/com/android/car/settings/users/UserGridRecyclerView.java
+++ b/src/com/android/car/settings/users/UserGridRecyclerView.java
@@ -253,7 +253,13 @@ public class UserGridRecyclerView extends PagedListView implements
 
                 // If the user selects Guest, start the guest session.
                 if (userRecord.mIsStartGuestSession) {
-                    mCarUserManagerHelper.startNewGuestSession(mGuestName);
+                    UserInfo currentGuestUser = mCarUserManagerHelper.findCurrentGuestUser();
+                    // If guest user already created, switch to it; otherwise, create a new guest user
+                    if (currentGuestUser != null) {
+                        mCarUserManagerHelper.switchToUser(currentGuestUser);
+                    } else {
+                        mCarUserManagerHelper.startNewGuestSession(mGuestName);
+                    }
                     return;
                 }
 
-- 
1.9.1

